pkg/sql/sem/tree/type_check.go

```diff
@@ -2564,6 +2564,15 @@ type typeCheckExprsState struct {
 	resolvableIdxs  intsets.Fast // index into exprs/typedExprs
 }
 
+func findFirstTupleIndex(exprs ...Expr) (index int, ok bool) {
+	for i, expr := range exprs {
+		if _, ok := expr.(*Tuple); ok {
```

Comment:
I know I'm a broken record here but we really need to stop inspecting the type of AST node to help determine the SQL type of expressions. All these cases are only partial fixes because they won't fix cases where the AST isn't Tuple but the SQL type produced is a tuple type.

This is fine for now, because it's a bit better than the current behavior. But maybe we should start labelling these places with TODOs that make it clear this is not the long-term solution? I don't want people to unknowingly repeat this pattern.